process/src/process.cpp

```diff
@@ -371,8 +371,137 @@ int create(lua_State* L)
     return lua_yield(L, 0);
 }
 
+static int envIndex(lua_State* L)
+{
+    const char* key = luaL_checkstring(L, 2);
+    char value[1024];
+    size_t size = sizeof(value);
+    int err = uv_os_getenv(key, value, &size);
+
+    if (err == UV_ENOBUFS)
+    {
+        char* buffer = (char*)malloc(size);
+        err = uv_os_getenv(key, buffer, &size);
+        if (err == 0)
+        {
+            lua_pushlstring(L, buffer, size);
+            free(buffer);
+            return 1;
+        }
+        free(buffer);
+    }
+    else if (err == UV_ENOENT)
+    {
+        lua_pushnil(L);
+        return 1;
+    }
+    else if (err != 0)
+    {
+        luaL_error(L, "Failed to get environment variable: %s", uv_strerror(err));
+        return 0;
+    }
+
+    lua_pushlstring(L, value, size);
+    return 1;
+}
+
+static int envNewindex(lua_State* L)
+{
+    const char* key = luaL_checkstring(L, 2);
+    int err;
+
+    if (lua_isnil(L, 3))
+    {
+        err = uv_os_unsetenv(key);
+    }
+    else
+    {
+        const char* value = luaL_checkstring(L, 3);
+        err = uv_os_setenv(key, value);
+    }
+
+    if (err != 0)
+    {
+        luaL_error(L, "Failed to set environment variable: %s", uv_strerror(err));
+    }
+
+    return 0;
+}
+
+typedef struct
+{
+    uv_env_item_t* items;
+    int count;
+    int index;
+} EnvIter;
```

Comment:
```suggestion
struct EnvIter
{
    uv_env_item_t* items;
    int count;
    int index;
}
```

We're writing C++ 😉

Reply:
still learning what a C++ is, thanks!

Reply:
it's funny because you didn't do this in the other one, but I guess you were reading more C sample code this time.